### ISSUE TITLE
mix ecto.migrate -n argument has no effect

### ISSUE THREAD

**Issue by bgentry** [elixir-ecto/ecto_sql#63]
### Environment

* Elixir version (elixir -v):
  ```
  Erlang/OTP 21 [erts-10.2] [source] [64-bit] [smp:16:16] [ds:16:16:10] [async-threads:1] [hipe] [dtrace]
  
  Elixir 1.7.4 (compiled with Erlang/OTP 21)
  ```
* Database and version (PostgreSQL 9.4, MongoDB 3.2, etc.): PostgreSQL v11.1
* Ecto version (mix deps): 3.0.2, ecto_sql 3.0
* Database adapter and version (mix deps): postgrex v0.14.1
* Operating system: macOS

### Current behavior

With multiple pending migrations, run `mix ecto.migrate -n 1`. Rather than only running a single migration, it runs all pending migrations.

### Expected behavior

The `-n 1` flag should work and cause Ecto to only migrate up by a single version, rather than running all pending migrations.

**Comment by josevalim** [comment]
Quick question: does it work with --step instead of -n?
-- 


*José Valim*
www.plataformatec.com.br
Skype: jv.ptec
Founder and Director of R&D

**Comment by bgentry** [comment]
> Quick question: does it work with --step instead of -n?

Nope, the argument is also ignored with `mix ecto.migrate --step 1` 🙁

**Comment by josevalim** [comment]
Hi @bgentry, unfortunately I cannot reproduce this using latest ecto and ecto_sql. I just started a new Phoenix app, created two different resources and did this:

```
~/OSS/phoenix/installer/demo[master %]$ mix ecto.migrate -n 1
[debug] QUERY OK source="schema_migrations" db=0.3ms
SELECT s0."version"::bigint FROM "schema_migrations" AS s0 FOR UPDATE []
[info] == Running 20181222002903 Demo.Repo.Migrations.CreateUsers.change/0 forward
[info] create table users
[info] == Migrated 20181222002903 in 0.0s
~/OSS/phoenix/installer/demo[master %]$ mix ecto.migrate -n 1
[debug] QUERY OK source="schema_migrations" db=0.8ms
SELECT s0."version"::bigint FROM "schema_migrations" AS s0 FOR UPDATE []
[info] == Running 20181222002912 Demo.Repo.Migrations.CreatePosts.change/0 forward
[info] create table posts
[info] == Migrated 20181222002912 in 0.0s
```

Have you tried using latest `ecto_sql` and checked if the problem persists? If so, can you prodice a way to reproduce it? Thanks.

**Comment by josevalim** [comment]
I have changed master to perform strict argument parsing, which we should have been doing along time ago:

```
$ mix ecto.dump -n 1
** (Mix) Could not invoke task "ecto.dump": 1 error found!
-n : Unknown option
```

Btw, you can change your alias to do this:

```elixir
def aliases do
  ["ecto.migrate": &migrate_and_dump/1]
end

defp migrate_and_dump(args) do
  Mix.Task.run "ecto.migrate", args
  Mix.Task.run "ecto.dump"
end
```

**Comment by bgentry** [comment]
@josevalim brilliant idea, thanks! Explicit rejection of invalid args will help prevent others from running into this class of problem. And also, thanks for the solution 😄
